Subject: ibmvnic: Fix incorrectly defined ibmvnic_request_map_rsp
 structure
From: Thomas Falcon <tlfalcon@linux.vnet.ibm.com>
X-Patchwork-Id: 779079

References: bsc#1045568
Patch-mainline: queued
Git-commit: 288ccb75b8fbf554d3fe21d1f91dbf489f10c6d4
Git-repo: https://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git

This reserved area should be eight bytes in length instead of four.
As a result, the return codes in the REQUEST_MAP_RSP descriptors
were not being properly handled.

Signed-off-by: Thomas Falcon <tlfalcon@linux.vnet.ibm.com>
Acked-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/net/ethernet/ibm/ibmvnic.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ibm/ibmvnic.h b/drivers/net/ethernet/ibm/ibmvnic.h
index 7e2300e..2d525c7 100644
--- a/drivers/net/ethernet/ibm/ibmvnic.h
+++ b/drivers/net/ethernet/ibm/ibmvnic.h
@@ -595,7 +595,7 @@ struct ibmvnic_request_map_rsp {
 	u8 cmd;
 	u8 reserved1;
 	u8 map_id;
-	u8 reserved2[4];
+	u8 reserved2[8];
 	struct ibmvnic_rc rc;
 } __packed __aligned(8);
 
